MiniApp/Classes/core/RealMiniApp.swift

```diff
@@ -106,59 +124,67 @@ internal class RealMiniApp {
-        self.miniAppStatus.setDownloadStatus(true, appId: appInfo.id, versionId: appInfo.version.versionId)
-        self.miniAppStatus.setCachedVersion(appInfo.version.versionId, for: appInfo.id)
+    func getMiniAppView(appInfo: MiniAppInfo, manifest: MiniAppManifest?, queryParams: String? = nil, completionHandler: @escaping (Result<MiniAppDisplayDelegate, Error>) -> Void, messageInterface: MiniAppMessageDelegate? = nil, adsDisplayer: MiniAppAdDisplayer? = nil) {
         isRequiredPermissionsAllowed(
-            appId: appInfo.id,
-            versionId: appInfo.version.versionId) { (result) in
+                appId: appInfo.id,
+                versionId: appInfo.version.versionId) { (result) in
             switch result {
             case .success:
-                    DispatchQueue.main.async {
-                        let miniAppDisplayProtocol = self.displayer.getMiniAppView(miniAppId: appInfo.id,
-                                                                                   versionId: appInfo.version.versionId,
-                                                                                   projectId: self.miniAppClient.environment.projectId,
-                                                                                   miniAppTitle: appInfo.displayName ?? "Mini app",
-                                                                                   queryParams: queryParams,
-                                                                                   hostAppMessageDelegate: messageInterface ?? self,
-                                                                                   adsDisplayer: adsDisplayer)
-                        completionHandler(.success(miniAppDisplayProtocol))
-                    }
+                DispatchQueue.main.async {
+                    let miniAppDisplayProtocol = self.displayer.getMiniAppView(miniAppId: appInfo.id,
+                                                                               versionId: appInfo.version.versionId,
+                                                                               projectId: self.miniAppClient.environment.projectId,
+                                                                               miniAppTitle: appInfo.displayName ?? "Mini app",
+                                                                               manifest: manifest,
+                                                                               queryParams: queryParams,
+                                                                               hostAppMessageDelegate: messageInterface ?? self,
+                                                                               adsDisplayer: adsDisplayer)
+                    self.miniAppStatus.setDownloadStatus(true, appId: appInfo.id, versionId: appInfo.version.versionId)
+                    self.miniAppStatus.setCachedVersion(appInfo.version.versionId, for: appInfo.id)
```

Comment:
These two lines should be moved before calling `isRequiredPermissionsAllowed()`, I guess this missed while doing conflict merge. Sorry about the confusion. 

The reason why we have moved them outside the block is to save the downloaded version details before checking `isRequiredPermissionsAllowed()`, so next time when create() is called, it will not download the same version again.

Reply:
That's good we have code reviews so you could catch it, thanks for spotting it ;)

Reply:
Looks like this change is missed @Climbatize